Improve core web vitals serving accessibility

shrinkray opened the pull request:
# Pull Request

## Description

fixes: #56 #55 #54 

- Address large contentful paint (LCP)
- Address missing favicons
- replace logo images with optimized SVG form
- designed event site logo as svg
- created new GDG group logo as svg
- refactor ResponsiveImage to use srcset form
- update date artwork for three new sizes, all artwork compressed with ImageMagic
- replace scaled icons with correct size pngs
- replace google logo with correct size
- replace dev team member images with cropped, equal size, and type
- set h2 font color, in some browsers the wrong color was returned

## Type of Change

<!-- Mark the relevant option with an "x" -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactoring
- [ ] Performance improvement
- [x] Accessibility improvement
- [x] Other (please describe): Core Web Vitals

## Related Issues

<!-- Link any related issues using "Fixes #123" or "Closes #123" -->

## Code Quality Checklist

<!-- Mark completed items with an "x" -->

- [x] Code follows the project's coding standards
- [x] Self-review of the code has been performed
- [ ] Code is properly commented where necessary
- [x] No console.log statements left in the code
- [x] No hardcoded values that should be configurable
- [ ] Error handling is implemented where appropriate

## Testing

<!-- Mark completed items with an "x" -->

- [x] Changes have been tested locally
- [x] All existing tests pass (if applicable)
- [ ] New tests have been added for new functionality (if applicable)
- [x] Manual testing has been performed
- [x] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

<!-- Mark completed items with an "x" -->

- [x] All interactive elements are keyboard accessible
- [x] Color contrast meets WCAG AA standards
- [x] Images have appropriate alt text
- [ ] Form labels are properly associated
- [x] Focus indicators are visible
- [ ] Screen reader compatibility has been tested

## Build & Deployment

<!-- Mark completed items with an "x" -->

- [x] Application builds successfully
- [x] No build warnings or errors
- [ ] Dependencies are up to date
- [x] No security vulnerabilities introduced

## Screenshots (if applicable)

<!-- Add screenshots to help explain your changes -->

## Additional Notes

- Site tested in Polypane browser and Google Chrome
- We can use a CSP for security to protect from XSS
- If helmet is added as a dependency, will utilize the metadata and favicons

---

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.

shrinkray committed:
fix: add aria label

- add aria-label to anchors where an image is enclosed

shrinkray committed:
feat: add meta content

- add logo images
- add meta to index.html
- work intended as placeholder until site is setup to enable other methods

shrinkray committed:
refactor: for mobile and responsive layout

- present embedded maps
- simpler dates
- containers stay within viewport

shrinkray committed:
refactor: responsive date images

- ResponsiveImage now enables real srcset values
- Is backwards compatible to original use
- browser will choose the image size
- user's screen resolution/pixel density
- viewport width
- sizes attribute guidance

shrinkray committed:
feat: add replacement images

- images compressed with ImageMagick 60% of exported size
- png and webp versions

shrinkray committed:
refactor: implement responsive image component

- phone with a 320px screen will load the 200px version
- tablet or small desktop might load the 300px version
- larger screens or high-DPI displays might load the 400px version
- meant to address CWV first contentful paint

shrinkray committed:
fix: file in wrong location

- move manifest to root
- file links to site icons for Android

shrinkray committed:
feat: add scaled icon png

- png icons were being scaled
- created actual size
- use ImageMagick to compress to 85%

shrinkray committed:
feat: add svg form of logo

- image optimized in SVGOMG
- image sized to usable value
- replaces image that was being scaled

shrinkray committed:
refactor: replace bitmap logo with svg

- tested in mobile
- pass in lighthouse tests
- no distortions

shrinkray committed:
refactor: update icon and text color

- replace icon with scaled version
- set text color on heading
- tested in Lighthouse

shrinkray committed:
refactor: image, heading, spacing

- use new icon
- set heading color
- add margin so icon has room

shrinkray committed:
refactor: replace logo with optimized

- add text color
- add preload attribute
- note fetchpriority attr not available in React

shrinkray committed:
refactor: replace with smaller logo

- performance fix

shrinkray committed:
refactor: replace images

- scale images
- compressed
- move to webp

shrinkray committed:
refactor: fix name

- update name based on whats app name

ShugKnight24 commented on src/layouts/LocationSection.jsx:
```diff
@@ -24,31 +24,25 @@ function LocationSection() {
                 className="block"
               >
-                Saturday: 9:00 AM - 9:00 PM
+                Saturday, November 22nd: 9:00 AM - 9:00 PM
```
Can you switch these times to be in line w/ #62

shrinkray replied:
Update now merged with 2025 branch

ShugKnight24 commented on index.html:
```diff
@@ -1,10 +1,62 @@
+    />
+    <link rel="icon" href="images/favicons/favicon-32x32.png" sizes="32x32" />
+    <link rel="icon" href="images/favicons/favicon-16x16.png" sizes="16x16" />
+    <link rel="apple-touch-icon" href="images/favicons/apple-touch-icon.png" />
```
This won't work with Vite

You'd either have to create a public directory, with images and favicons subdirectories with these assets and update the path in site.webmanifest.json or use them as variables in the React side of things and use their way of adding metedata to the document - https://react.dev/reference/react-dom/components/meta

I'd also recommend adding `<link rel="icon" href="/images/favicons/favicon.ico" type="image/x-icon" />` at the top level in case the browser doesn't support a svg favicon

My personal opinion is these favicons don't look great on a transparent background and should likely be on a black background or w/e the group prefers

shrinkray replied:
- public directory added to root
- favicons in question moved to it
- paths updated

ShugKnight24 replied:
can confirm, favicons are loaded and work correctly now when previewing

ShugKnight24 commented on site.webmanifest.json:
```diff
@@ -0,0 +1,22 @@
+      "type": "image/png"
+    },
+    {
+      "src": "src/assets/images/favicons/android-chrome-512x512.png",
```
update these paths if you go the public directory route

shrinkray replied:
@ShugKnight24 based on your based comments, I will setup a public folder for the favicons. And I will do this item mentioned below for content updates; plus the other things of tailwind variety.

ShugKnight24 commented on src/layouts/LandingSection.jsx:
```diff
@@ -2,29 +2,39 @@
-            pngSrc={DevFestLogoPng}
-            alt="Michigan DevFest 2025 logo with Detroit skyline and developer graphics"
+    <section id="landing" className="mt-16 bg-black">
+      <div className="mx-auto flex max-w-screen-lg flex-wrap ">
```
NIT: EOL space

I'm personally not a fan of `max-w-screen-lg` though I'll defer to the group

shrinkray replied:
- style `max-w-[1554px]` performed best
- open to other solutions

ShugKnight24 replied:
Good fix - My issue w/ `max-w-screen-lg` was this look

<img width="1540" height="694" alt="Screenshot 2025-10-09 at 5 14 53 PM" src="https://github.com/user-attachments/assets/eafef68f-e84a-434d-8a29-5b8123715e15" />

`max-w-[1554px]` solves it and makes it look much better imo

<img width="1550" height="605" alt="Screenshot 2025-10-09 at 5 16 01 PM" src="https://github.com/user-attachments/assets/e39bc48c-b6c0-4a7d-94de-db245ca035ce" />

ShugKnight24 commented on src/layouts/LandingSection.jsx:
```diff
@@ -2,29 +2,39 @@
+              sizes="(max-width: 570px) 100vw, 570px"
               alt="November 21-22, 2025 event dates with red circular badge"
             />
 
```
@JustinBeaudry made a couple of good updates in his PR which he closed to the text in the landing section. Can you make those adjustments here

```
  The Detroit, Ann Arbor, Toledo, UofM Dearborn and Windsor GDGs are
  teaming up with Compass Detroit to bring you the 11th Michigan
  DevFest on November 21-22, 2025.
  <br />
  <br />
  Join us on Friday, November 21st for an Agentic AI Hackathon at
  Wayne State University&apos;s Saint Andrew&apos;s Memorial
  Episcopal Church in Detroit, followed by DevFest sessions on
```

shrinkray replied:
Finished updates adding this snippet.

ShugKnight24 reviewed (changes requested):
Good PR @shrinkray

Main request for changes is for the setup of the favicons in Vite

Can you also make the time adjustments to the LandingSection and add in @JustinBeaudry's text updates

shrinkray committed:
Merge branch '2025' into shrinkray/accessibility

shrinkray committed:
refactor: move favicons to public folder

- create new public folder in root
- move favicon images inside folder
- reconnect instances to public path

shrinkray committed:
fix: change tailwind class

- per request from ShugKnight24

shrinkray committed:
fix: use rewritten text

- per JustinBeaudry

JustinBeaudry reviewed (approved):
From my perspective, I'm not seeing any issues. This look good to me, but I would prefer we wait for @ShugKnight24's review

ShugKnight24 reviewed (approved):
LGTM - thanks for the updates @shrinkray
